shared/balance/balance.go

```diff
@@ -134,43 +159,57 @@ func UtxoBasedBalanceReq(endpoints []string) (string, error) {
 	case <-time.After(2 * time.Second):
 		return "", errors.New("Bad request")
 	}
-
 }
 
 // ETH based
 func GetEtherBalance(address string) (string, error) {
 
-	currency := os.Getenv("blockchain")
+	ethClient := ethrpc.New(estorage.EndpointForReq.Get())
 
-	endpoints := estorage.EndpointsFromDB.Get(currency).Addresses
-	endpoints = append(endpoints, os.Getenv("main-api"))
-
-	result, err := EtherBalanceReq(endpoints, address)
+	res, err := ethClient.EthGetBalance(address, "latest")
 	if err != nil {
-		return "", err
+		balance, err := EtherBalanceReq(address)
+		if err != nil {
+			return "", err
+		}
+
+		return balance, nil
 	}
 
-	return result, nil
+	return res.String(), nil
 
 }
 
 func GetTokenBalance(userAddress, smartContractAddress string) (string, error) {
-	currency := os.Getenv("blockchain")
-
-	endpoints := estorage.EndpointsFromDB.Get(currency).Addresses
-	endpoints = append(endpoints, os.Getenv("main-api"))
 
-	var balance string
+	ethClient, err := ethclient.Dial(estorage.EndpointForReq.Get())
+	if err != nil {
+		return "", err
+	}
 
-	balance, err := TokenBalanceReq(endpoints, userAddress, smartContractAddress)
+	instance, err := abi.NewToken(common.HexToAddress(smartContractAddress), ethClient)
 	if err != nil {
 		return "", err
 	}
 
-	return balance, nil
+	res, err := instance.BalanceOf(nil, common.HexToAddress(userAddress))
+	if err != nil {
+		balance, err := TokenBalanceReq(userAddress, smartContractAddress)
+		if err != nil {
+			return "", err
```

Comment:
can we log errors to Elastic Search @maxUo what do you think?

Reply:
I think we can tell about it on the next sprint